File: 2475. Number of Unequal Triplets in Array/Solution.cpp
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    int unequalTriplets(vector<int>& nums)
    {
        unordered_map<int, int> cnt;
        for (int x : nums)
            cnt[x]++;
        int res = 0, n = nums.size(), t = 0;
        for (const auto& [_, v] : cnt) {
            res += t * v * (n - t - v);
            t += v;
        }
        return res;
    }
};

File: 2475. Number of Unequal Triplets in Array/Solution.py
from collections import Counter
from typing import List


class Solution:
    def unequalTriplets(self, nums: List[int]) -> int:
        res = t = 0
        n = len(nums)
        for v in Counter(nums).values():
            res += t * v * (n - t - v)
            t += v
        return res
